### ISSUE TITLE
Exposing logger sinks

### ISSUE THREAD

**Issue by dhalinar** [gabime/spdlog#286]
Are there any (immediately obvious) problems exposing a logger's sinks, with say a 'get_sinks()' function?

I'm imaging some object Foo with logger "A" and a stdout sink.
Then some Bar wants to share Foo's logger's sink but with a logger named "B".

```
Foo foo;
auto fooLogger = spdlog::stdout_logger_st("foo logger");
foo.setLogger(fooLogger);

Bar bar;
std::vector<spdlog::sink_ptr> sinks = fooLogger.get_sinks();
auto barLogger = spdLog::create("bar logger", sinks.begin(), sinks.end());
bar.setLogger(barLogger);
```

The example is a bit contrived, but what I'm going for is having different 'logger' names in a shared sink.

**Comment by gabime** [comment]
yes, seems a reasonable feature to have.

**Comment by gabime** [comment]
merged pull #289
